### PR TITLE
Add non-evm namespaces in mobile

### PR DESCRIPTION
**Pull request by onghwan**
* This PR adds optional namespaces for non-evm chains support. Since core web integrated wagmi and it only supports EVM for now, it throws an error when we add these non-EVM namespaces in the dapp. This is a temporary fix until core web supports these namespaces.

### CODE REVIEW

**Review by ruijialin-avalabs — DISMISSED** [review]

File: packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/namespaces.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      RpcMethod.AVALANCHE_REMOVE_CONTACT,
+      RpcMethod.AVALANCHE_SELECT_ACCOUNT,
+      RpcMethod.AVALANCHE_SET_DEVELOPER_MODE,
+      RpcMethod.AVALANCHE_UPDATE_CONTACT,
+      RpcMethod.AVALANCHE_SEND_TRANSACTION,
+      RpcMethod.AVALANCHE_SIGN_TRANSACTION,
+      RpcMethod.AVALANCHE_GET_ADDRESSES_IN_RANGE,
+      RpcMethod.BITCOIN_SEND_TRANSACTION,
+      RpcMethod.AVALANCHE_SIGN_MESSAGE
+    ],
```
Comment (atn4z7) [inline packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/namespaces.ts:31]:
```suggestion
    methods: [
      RpcMethod.AVALANCHE_SEND_TRANSACTION,
      RpcMethod.AVALANCHE_SIGN_TRANSACTION,
      RpcMethod.BITCOIN_SEND_TRANSACTION,
      RpcMethod.AVALANCHE_SIGN_MESSAGE
    ],
```

Comment (atn4z7) [reply packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/namespaces.ts:31]: let's also remove these methods from the CORE_ONLY_METHODS array

File: packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/wc_sessionRequest.ts [hunk 2]
```diff
@@ -200,7 +201,14 @@ class WCSessionRequestHandler implements RpcRequestHandler<WCSessionProposal> {
...
 
     const normalizedRequired = normalizeNamespaces(requiredNamespaces)
-    const normalizedOptional = normalizeNamespaces(optionalNamespaces)
+    const normalizedOptional = normalizeNamespaces({
+      ...optionalNamespaces,
+      // add optional namespaces for non-evm chains support
+      // since core web integrated wagmi and it only supports EVM for now,
+      // it throws an error when we add these non-EVM namespaces in the dapp.
+      // This is a temporary fix until core web supports these namespaces
+      ...NONEVM_OPTIONAL_NAMESPACES
```
Comment (atn4z7) [inline packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/wc_sessionRequest.ts:210]: this should only be added for core apps

File: packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/wc_sessionRequest.ts [hunk 3]
```diff
@@ -200,7 +201,14 @@ class WCSessionRequestHandler implements RpcRequestHandler<WCSessionProposal> {
     const { proposer, requiredNamespaces, optionalNamespaces } = params
 
     const normalizedRequired = normalizeNamespaces(requiredNamespaces)
```
Comment (atn4z7) [inline packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/wc_sessionRequest.ts:203]: can we add them to requiredNamespaces instead?

File: packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/namespaces.ts (hunk 1 above)
Comment (ruijialin-avalabs) [reply packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/namespaces.ts:31]: should RpcMethod.BITCOIN_SEND_TRANSACTION be in another namespace?  bip155?

Comment (onghwan) [reply packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/namespaces.ts]: fixed.

Comment (onghwan) [reply packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/namespaces.ts]: I believe @atn4z7 will move it to bip155 if he need it.

File: packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/wc_sessionRequest.ts (hunk 2 above)
Comment (onghwan) [reply packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/wc_sessionRequest.ts]: fixed it to add NONEVM_OPTIONAL_NAMESPACES only when isCoreDomain is true

File: packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/wc_sessionRequest.ts (hunk 3 above)
Comment (onghwan) [reply packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/wc_sessionRequest.ts:205]: I tried, but I got this error. 

Simulator Screenshot - iPhone 15 Pro - 2024-08-13 at 12 31 34

so I guess this workaround only works with optional namespaces injected by the client.

**Review by atn4z7 — APPROVED** [review]

**Pull request merged**